### ISSUE TITLE
AUR pkg incorrectly detected as devel

### ISSUE THREAD

**Issue by sum01** [Jguer/yay#691]
#### Affected Version

yay v8.1115 - libalpm v11.0.1
#### Issue
[cquery](https://aur.archlinux.org/packages/cquery/) is detected as a devel package because it uses git in its sources, but they are tags.

Each time I do `yay -Syu --devel` it tries to upgrade it, even though I already have the latest version, causing failures.

---

This would be resolved if git sources were not marked devel when they point at a specific tag/commit, since they won't change upon rebuilds of the PKGBUILD.

**Comment by Morganamilo** [comment]
Yay should already ignore urls with commit= or tag=. I'll take a look into whats broken later.

**Comment by Morganamilo** [comment]
Can't say I can reproduce it. Can you show me the output of an upgrade and the contents of `~/.cache/yay/vcs.json`?

**Comment by sum01** [comment]
What I specifically did was already have Cquery installed, then when I went to upgrade it says...
```
:: 1 Packages to upgrade.
1  devel/cquery  v20180718-4 -> latest-commit
```

Which I was curious and pressed show diff, which it then failed to find a diff (since there wasn't one). I tried again, but this time didn't show diff, and it went through the steps and told me it was reinstalling (since I already have the newest).

I guess it's just weird behaviour.

[vcs.json.zip](https://github.com/Jguer/yay/files/2362486/vcs.json.zip)

**Comment by Morganamilo** [comment]
It would appear it didn't always use specific commits https://aur.archlinux.org/cgit/aur.git/commit/?h=cquery&id=af5c9b31387a4d01adcba89beb2e63e0f483b4b5. So I would guess yay picked up on that and there's not really a mechanism in place for yay to forget it.

You can either delete the entry from vcs.json or uninstall then install the package. Yay should handle this though.

**Comment by Morganamilo** [comment]
Actually I gave the pkgbuild a proper look, I noticed they've done `?commit=` instead of `#commit=`. So actually this package does not track a specific commit and does indeed pull from master.

>and it went through the steps and told me it was reinstalling (since I already have the newest).

It did actually pull a version using newer commits. But as the pkgver() function is not set up to generate a never version the pkgver stayed the same and it never rebuild. Passing --reinstall would have indeed built a newer version even if the version number stayed the same.
